src/aiida/cmdline/commands/cmd_calcjob.py

```diff
@@ -348,3 +348,80 @@ def get_remote_and_path(calcjob, path=None):
         f'nor does its associated process class `{calcjob.process_class.__class__.__name__}`\n'
         'Please specify a path explicitly.'
     )
+
+
+@verdi_calcjob.command('dump')
+@arguments.CALCULATION('calcjob', type=CalculationParamType(sub_classes=('aiida.node:process.calculation.calcjob',)))
+@options.PATH()
+@options.NO_NODE_INPUTS()
+@options.INCLUDE_ATTRIBUTES()
+@options.INCLUDE_EXTRAS()
+@options.USE_PRESUBMIT()
+@options.OVERWRITE()
+def dump(
+    calcjob,
+    path,
+    no_node_inputs,
+    include_attributes,
+    include_extras,
+    use_presubmit,
+    overwrite,
+) -> None:
+    """Dump files involved in the execution of a `CalcJob`.
+
+    Child simulations/workflows (also called `CalcJob`s and `WorkChain`s in AiiDA jargon) run by the parent workflow are
+    contained in the directory tree as sub-folders and are sorted by their creation time. The directory tree thus
+    mirrors the logical execution of the workflow, which can also be queried by running `verdi process status
+    <pk>` on the command line.
+
+    By default, input and output files of each simulation can be found in the corresponding "raw_inputs" and
+    "raw_outputs" directories (the former also contains the hidden ".aiida" folder with machine-readable job execution
+    settings). Additional input files (depending on the type of calculation) are placed in the "node_inputs".
+
+    When using the `--use-presubmit` command line option, the folder created for each individual simulation should, in
+    principle, allow for direct resubmission, as it mirrors the (remote) folder that was created by AiiDA to execute the
+    job. However, this option requires the relevant AiiDA plugin to be installed, so it is disabled by default. Also
+    note that intermediate files might be missing, so for a multi-step workflow, each step would still have to be run
+    separately.
```

Comment:
Right, I forgot to add that to the caveats, and just assumed it implicitly. We could only allow for that option in the case of `verdi calcjob dump` where it makes more sense, though, as I'm typing this, I realize it's also not true there - for any `CalcJob` that's not only run by itself, or is the first step of a `WorkChain`.

In the end, there is absolutely no way we can guarantee that this generally works. Now, do we still want to provide the option to users despite this problem, as a shortcut for dumping the files in a directory structure they are used to from running their codes without AiiDA, or we fully remove it due to this shortcoming. I still think it could be a useful feature, as otherwise we force them to always move/copy/rename their dumped files around if they want to manually submit a calculation they dumped from AiiDA, which could be annoying.

Reply:
Actually, for this use-case, we already have a feature that allows this, it is called the `dry_run`. Essentially when running your calculation you just add `metadata.dry_run = True` to the inputs and the input files are written to a directory in the CWD. Maybe that would already suffice?

Reply:
Hm, for more advanced users that's a good option. Though, I think a main use case for this feature is people who don't want to run calculations - they just get an archive, dump it, and explore it like they are used to on the file system. Could we still use the `dry_run` option in our current setup? If not, I'd say we shelf the `--use-presubmit` flag, in favor of getting the code in shape for the upcoming release.